### PR TITLE
Improve support for configuration property nesting

### PR DESCRIPTION
**Pull request by graemerocher**
Currently injecting nested lists and maps of records or interfaces doesn't work and by default tries to bind the value. This resolves that by using dependency injection for configuration property entries that are beans by default.

Also improves the documentation generation to use a javadoc parser to parse only the relevant parts of the javadoc

### CODE REVIEW

**Review by dstepanov — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

File: core-processor/build.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ dependencies {
     api project(":aop")
     api libs.asm.tree
     api libs.bundles.asm
+    implementation 'com.github.javaparser:javaparser-symbol-solver-core:3.24.7'
```
Comment (graemerocher) [inline core-processor/build.gradle]: this should  be moved to the version catalog after @melix PR is merged

**Comment by graemerocher** [comment]
Test failures are unrelated

File: core-processor/build.gradle [hunk 2]
```diff
@@ -7,6 +7,7 @@ dependencies {
     api project(":aop")
     api libs.asm.tree
     api libs.bundles.asm
+    api 'com.github.javaparser:javaparser-symbol-solver-core:3.24.7'
```
Comment (sdelamo) [inline core-processor/build.gradle:10]: we are ok adding this extra dependency because core-processor will be a `compileOnly` dependency right?

**Review by sdelamo — APPROVED** [review]

File: core-processor/build.gradle (hunk 2 above)
Comment (graemerocher) [reply core-processor/build.gradle:10]: correct

Comment (dstepanov) [reply core-processor/build.gradle:10]: I think it should be in libs.versions.toml

Comment (graemerocher) [reply core-processor/build.gradle:10]: it should be but as per my comment I didn't want to conflict with @melix's PR so left it like that so we can fix it later

**Pull request merged**